### PR TITLE
fix(umd): prevent inclusion of Vue in bundle

### PR DESCRIPTION
**Pull request by Haroenv**
This is fixed in two spots:

1. add Vue to externals for the UMD (used in Panel)
2. use the Vue object passed to the install function instead of importing it ourselves (in the plugin)

Furthermore this also fixes the plugin which needlessly exposes all widgets (you can already import them easily, since the default export is being put on the window)

### CODE REVIEW

**Comment by francoischalifour** [comment]
This warning appears while building.

```
src/instantsearch.umd.js → dist/vue-instantsearch.js...
┌────────────────────────────────────────────┐
│                                            │
│   Destination: dist/vue-instantsearch.js   │
│   Bundle Size:  224.06 KB                  │
│   Minified and Gzipped Size:  57.93 KB     │
│                                            │
└────────────────────────────────────────────┘
(!) Missing global variable name
Use output.globals to specify browser global variable names corresponding to external modules
vue (guessing 'Vue')
```

**Comment by Haroenv** [comment]
good call, I didn't have it earlier, but can replicate now @francoischalifour

**Comment by Haroenv** [comment]
by the way, nice effect of this:

## before

```
src/instantsearch.umd.js → dist/vue-instantsearch.js...
┌────────────────────────────────────────────┐
│                                            │
│   Destination: dist/vue-instantsearch.js   │
│   Bundle Size:  284.17 KB                  │
│   Minified Size:  281.05 KB                │
│   Gzipped Size:  79.27 KB                  │
│                                            │
└────────────────────────────────────────────┘
```

## after

```
src/instantsearch.umd.js → dist/vue-instantsearch.js...
┌────────────────────────────────────────────┐
│                                            │
│   Destination: dist/vue-instantsearch.js   │
│   Bundle Size:  224.09 KB                  │
│   Minified Size:  221.15 KB                │
│   Gzipped Size:  57.93 KB                  │
│                                            │
└────────────────────────────────────────────┘
```

**Comment by francoischalifour** [comment]
Nice decrease!

File: src/plugin.js [hunk 1]
```diff
@@ -1,12 +1,11 @@
 /* eslint import/namespace: ['error', { allowComputed: true }]*/
-import Vue from 'vue';
 
 import * as widgets from './widgets';
 
-export const plugin = Object.assign({}, widgets, {
-  install() {
+export const plugin = {
+  install(Vue) {
```
Comment (samouss) [inline src/plugin.js]:
```suggestion
  install(localVue) {
```

We could rename it to indicate that it's clearly not the global one, it avoids a bit of confusion.

**Comment by samouss** [comment]
It's better without Vue embed!

**Pull request merged**